#goal definition
char[] voice_output
---
#result definition
int32[] sequence
---
#feedback
int32[] sequence